kubernetes/provider.go

```diff
@@ -101,6 +101,33 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("KUBE_LOAD_CONFIG_FILE", true),
 				Description: "Load local kubeconfig.",
 			},
+			"exec": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"api_version": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"command": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"env": {
+							Type:     schema.TypeMap,
```

Comment:
While the Terraform Provider SDK does not explicitly error when not providing the `Elem` for `Type: schema.TypeMap`, we generally recommend adding it, e.g. `Elem: &schema.Schema{Type: schema.TypeString},`